### DISCUSSION TITLE
Serve js, css and fonts locally, i.e. not from CDN

### DISCUSSION THREAD

**Discussion by violoncelloCH** [heyform/heyform#144] [Ideas]
It would be nice to have resources (js, css and fonts) being served locally by Heyform and not through various CDNs. This is because establishing connections, to google, cloudflare etc. leaks information to these third parties.
At least having an option to disable embedding the remote resources should be available.

**Comment by iMuFeng** [comment]
In our previous code commit, we replaced the CDN resources with local ones. https://github.com/heyform/heyform/issues/84

**Reply by violoncelloCH** [reply]
Thanks for your reply.
While I see that what has been done in #84 is related to my question, there are still multiple things loaded from cloudflare, google etc:
- https://github.com/heyform/heyform/blob/bf867b5c146023583b3c1f9a26b18881ffb38ea3/packages/webapp/index.html  
  here lines 6-8 initiate dns requests for those different domains; then in lines 23 - 28 resources are loaded from cloudflare and google.
- https://github.com/heyform/heyform/blob/bf867b5c146023583b3c1f9a26b18881ffb38ea3/packages/form-renderer/src/theme.ts#L77  
  line 77 here also makes the browser initiate requests to google
- potentionally more I did not yet look at...

It would be cool if those could be avoided by direcly serving the required fonts, css etc from heyform.

**Comment by dearroy** [comment]
Just an update:

You can now switch to self-hosted fonts by setting `ENABLE_GOOGLE_FONTS=false`.
